### PR TITLE
ported response.js

### PR DESCRIPTION
**Pull request by raylin**

### CODE REVIEW

File: src/response.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  send(...strings) {
+    this.runWithMiddleware('send', { plaintext: true }, ...strings);
+  }
+
+  /**
+   * Public: Posts an emote back to the chat source
+   * 
+   * @params {...string} strings - One or more strings to be posted. The order of these strings
+   *           should be kept intact.
+   */
```
Comment (gasolin) [inline src/response.js:43]: `* Returns nothing.`

File: src/response.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      user: this.message.user,
+      message: this.message
+    }
+  }
+
+  /**
+   * Public: Posts a message back to the chat source
+   * 
+   * @params {...string} strings - One or more strings to be posted. The order of these strings
+   *           should be kept intact.
```
Comment (gasolin) [inline src/response.js:28]: `* Returns nothing.`

File: src/response.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+   */
+  emote(...strings) {
+    this.runWithMiddleware('emote', { plaintext: true }, ...strings);
+  }
+
+  /**
+   * Public: Posts a message mentioning the current user.
+   * 
+   * @params {...string} strings - One or more strings to be posted. The order of these strings
+   *           should be kept intact.
```
Comment (gasolin) [inline src/response.js:52]: `* Returns nothing.`

File: src/response.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+   */
+  reply(...strings) {
+    this.runWithMiddleware('reply', { plaintext: true }, ...strings);
+  }
+
+  /**
+   * Public: Posts a topic changing message
+   * 
+   * @params {...string} strings - One or more strings to set as the topic of the
+   *           room the bot is in.
```
Comment (gasolin) [inline src/response.js:64]: `* Returns nothing.`

File: src/response.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+   */
+  topic(...strings) {
+    this.runWithMiddleware('topic', { plaintext: true }, ...strings);
+  }
+
+  /**
+   * Public: Play a sound in the chat source
+   *
+   * @params {...string} strings - One or more strings to be posted as sounds to play. The order of
+   *           these strings should be kept intact.
```
Comment (gasolin) [inline src/response.js:76]: `* Returns nothing.`

File: src/response.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+   */
+  play(...strings) {
+    thisrunWithMiddleware('play', ...strings);
```
Comment (gasolin) [inline src/response.js]: `this.runWithMiddleware`

File: src/response.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+  }
+
+  /**
+   * Public: Posts a message in an unlogged room
+   *
+   * @params {...string} strings - One or more strings to be posted. The order of these strings
+   *           should be kept intact.
```
Comment (gasolin) [inline src/response.js:88]: `* Returns nothing.`

File: src/response.js [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+   * send the request.
+   *
+   * @return {object} A ScopedClient instance.
+   */
+  http(url, options) {
+    return this.robot.http(url, options);
+  }
+}
+
+export {Response as default}
```
Comment (gasolin) [inline src/response.js]: let's use common js module for the first version?

`module.exports = Response`

**Comment by gasolin** [comment]
related to https://github.com/gasolin/webbybot/issues/13

**Comment by gasolin** [comment]
Thanks for PR! please port the test cases to `test/response_test.js` as well

**Comment by raylin** [comment]
It seems no response_test.js in test folder. Perhaps it is covered by other tests?

**Comment by gasolin** [comment]
OK that's merge it!

**Pull request merged**